topdown/http.go

```diff
@@ -67,17 +68,51 @@ func builtinHTTPSend(bctx BuiltinContext, args []*ast.Term, iter func(*ast.Term)
 		return handleBuiltinErr(ast.HTTPSend.Name, bctx.Location, err)
 	}
 
+	usePersistentCache, err := getBoolValFromReqObj(req, "cache")
```

Comment:
Nit: relying on booleans like this for simple conditionals is fine. Where it gets a bit questionable is when you have (perhaps multiple) complex code paths checking these flags. Another way to organize the code would be to put the logic behind an interface. E.g., we could have an `HTTPRequestExecutor` interface with two (for now) implementations:

1. Current intra-query caching
2. New inter-query caching

Then we just construct the right executor and run it. No need for interleaved code paths.

Reply:
Defined a new `HTTPRequestExecutor ` interface and added implementations for the `inter-query` and `intra-query` cache.